src/plugins/CoreLoadPlugin.ts

```diff
@@ -72,17 +159,141 @@ export default class DojoLoadPlugin {
+											while (index > 0) {
+												const entry = callPath[ index-- ];
+
+												if (entry.type === 'CallExpression') {
+													if (entry.callee.type === 'MemberExpression' && entry.callee.property.type === 'Identifier' && entry.callee.property.name === 'define') {
+														foundDefineCall = true;
+														break;
+													}
+												}
+											}
+
+											/*
+											We only want to process calls that were made inside of a `registry.define` call.
+											 */
+											if (!foundDefineCall) {
+												return;
+											}
+
+											/*
+											 Find the containing function of the expression. We'll want this whole
+											 function to be wrapped in the require.ensure
+											 */
+											let fnExpression = callPath.pop();
+											while (fnExpression && fnExpression.type !== 'FunctionExpression') {
+												fnExpression = callPath.pop();
+											}
+
+											/*
+											 The require.ensure plugin expects you to actually be calling require.ensure,
+											 which has a signature like require.ensure([], function() { }). We need to mock
+											 the plugin has a hard coded check on 'expression.arguments[1]' to get a handle
+											 to the actual function we want to ensure, so we need to make a pretend expression
+											 that looks like a require.ensure call.
+											 */
+											const temp = {
+												type: 'CallExpression',
+												arguments: [
+													{},
+													fnExpression
+												],
+												range: fnExpression.range
+											};
+
+											/*
+											 Find an appropriate chunk name (null is an appropriate chunk name).
+											 */
+											let chunkName = path.basename(absolutePath);
+
+											const applicableNames = Object.keys(chunkNames).filter(name => {
+												return chunkNames[ name ].test(<string> second.value);
+											});
+
+											if (applicableNames.length > 0) {
+												chunkName = applicableNames[ 0 ];
+											}
+
+											/*
+											 Create the require.ensure block
+											 */
+											const dep = new RequireEnsureDependenciesBlock(temp, fnExpression, chunkName, null, parser.state.module, fnExpression.loc);
+
+											const old = parser.state.current;
+											parser.state.current = dep;
+
+											/*
+											 We add our one dependency to the [] in the ensure
+											 */
+											(<any> parser).inScope([], () => {
+												const edep = new RequireEnsureItemDependency(second.value, second.range);
+												edep.loc = dep.loc;
+												dep.addDependency(edep);
+											});
+
+											/*
+											 By default, the require.ensure is not going to execute when we want. We wrap it in a function block
+											 to control the execution.
+											 */
+											(<any> old).addDependency(new ConstDependency('function() { return (', fnExpression.range[0]));
```

Comment:
As you have to cast all the usages would it better to `const old: any = parser.state.current;` (if that works) and then just cast back to the current type when it gets reassigned? `parser.state.current = <CurrentType> old;`?